src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/UseStringEqualsOverStringCompare.Fixer.cs

```diff
@@ -0,0 +1,218 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Operations;
+
+using Resx = Microsoft.NetCore.Analyzers.MicrosoftNetCoreAnalyzersResources;
+using RequiredSymbols = Microsoft.NetCore.Analyzers.Runtime.UseStringEqualsOverStringCompare.RequiredSymbols;
+
+namespace Microsoft.NetCore.Analyzers.Runtime
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, LanguageNames.VisualBasic), Shared]
+    public sealed class UseStringEqualsOverStringCompareFixer : CodeFixProvider
+    {
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var document = context.Document;
+            var token = context.CancellationToken;
+            var semanticModel = await document.GetSemanticModelAsync(token).ConfigureAwait(false);
+
+            if (!RequiredSymbols.TryGetSymbols(semanticModel.Compilation, out var symbols))
+                return;
+
+            var root = await document.GetSyntaxRootAsync(token).ConfigureAwait(false);
+
+            if (semanticModel.GetOperation(root.FindNode(context.Span, getInnermostNodeForTie: true), token) is not IBinaryOperation violation)
+                return;
+
+            //  Get the replacer that applies to the reported violation. Bail out if no replacer 
+            //  matches the reported violation.
+            var replacer = GetOperationReplacers(symbols).FirstOrDefault(x => x.IsMatch(violation));
+            if (replacer is null)
+                return;
+
+            var codeAction = CodeAction.Create(
+                Resx.UseStringEqualsOverStringCompareCodeFixTitle,
+                CreateChangedDocument,
+                nameof(Resx.UseStringEqualsOverStringCompareCodeFixTitle));
+            context.RegisterCodeFix(codeAction, context.Diagnostics);
+            return;
+
+            //  Local functions
+
+            async Task<Document> CreateChangedDocument(CancellationToken token)
+            {
+                var editor = await DocumentEditor.CreateAsync(document, token).ConfigureAwait(false);
+                var replacementNode = replacer.CreateReplacementExpression(violation, editor.Generator);
+                editor.ReplaceNode(violation.Syntax, replacementNode);
+
+                return editor.GetChangedDocument();
+            }
+        }
+
+        public override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(UseStringEqualsOverStringCompare.RuleId);
```

Comment:
nit: Public properties usually go before public methods. It also makes it easier to tell what diagnostics this fixes (in case you couldn't figure it out from the name)

Reply:
Fixed.